common/encoding.go

```diff
@@ -34,35 +35,37 @@ func (a *Address) Encode() (string, error) {
 	return bech32.EncodeM(a.HRP, grp)
 }
 
-// DecodeAddress parses a bech32m encoded address string and returns an Address object
+// DecodeAddress parses a bech32m encoded address string and returns an Address struct.
 func DecodeAddress(addr string) (*Address, error) {
 	if len(addr) == 0 {
-		return nil, fmt.Errorf("address is empty")
+		return nil, fmt.Errorf("mssing address")
 	}
 
 	prefix, buf, err := bech32.DecodeNoLimit(addr)
 	if err != nil {
 		return nil, err
 	}
 	if prefix != Bitcoin.Addr && prefix != BitcoinTestNet.Addr && prefix != BitcoinRegTest.Addr {
-		return nil, fmt.Errorf("invalid prefix")
+		return nil, fmt.Errorf("unknown prefix")
 	}
 	grp, err := bech32.ConvertBits(buf, 5, 8, false)
 	if err != nil {
 		return nil, err
 	}
 
-	serverKey, err := schnorr.ParsePubKey(grp[:32])
+	version := uint32(grp[0])
+	serverKey, err := schnorr.ParsePubKey(grp[1:33])
```

Comment:
I'm realizing that maybe it's not safe to access `grp` without checking its length first ? it may panic if someone pass a smaller string ?